httpclient5/src/main/java/org/apache/hc/client5/http/utils/Base64.java

```diff
@@ -0,0 +1,223 @@
+     */
+    public static byte[] decodeBase64(final byte[] base64) {
+        if (null == base64) {
+            return null;
+        }
+
+        return decodeOrEmptyBytes(UrlByteRemapper.toStandardUrlEncoding(base64));
+    }
+
+    /**
+     * Decodes a Base64 String into octets.
+     * <p>
+     * <b>Note:</b> this method seamlessly handles data encoded in URL-safe or normal mode.
+     */
+
+    public static byte[] decodeBase64(final String base64) {
+        if (null == base64) {
+            return null;
+        }
+
+//        The acceptable Base64 alphabets have the same encodings in ASCII, ISO_8859_1, and UTF-8.
+//        ISO_8859_1 is used here because it matches the choice in the Java 8 Base64 implementation, and is
+//        in principle a tiny bit faster than the others to convert in versions of Java that support "compact strings".
+//        Any inputs outside the accepted values will be invalid in any encoding.
+
+        final byte[] bytes = base64.getBytes(StandardCharsets.ISO_8859_1);
+        UrlByteRemapper.replaceUrlSafeBytes(bytes);
+
+        return decodeOrEmptyBytes(bytes);
+    }
+
+    /**
+     * Encodes binary data using the base64 algorithm but does not chunk the output.
+     */
+
+    public static byte[] encodeBase64(final byte[] bytes) {
+        if (null == bytes) {
+            return null;
+        }
+
+        return getEncoder().encode(bytes);
+    }
+
+    /**
+     * Encodes binary data using the base64 algorithm but does not chunk the output.
+     */
+
+    public static String encodeBase64String(final byte[] bytes) {
+        if (null == bytes) {
+            return null;
+        }
+
+        return getEncoder().encodeToString(bytes);
+    }
+
+    /**
+     * Encode bytes to their Base64 form, using specifications from this codec instance
+     */
+
+    public byte[] decode(final byte[] base64) {
+        if (null == base64) {
+            return null;
+        }
+
+        return decodeOrEmptyBytes(UrlByteRemapper.toStandardUrlEncoding(base64));
+    }
+
+    /**
+     * Encode bytes to their Base64 form, using specifications from this codec instance
+     */
+    public byte[] encode(final byte[] value) {
+        if (null == value) {
+            return null;
+        }
+        return getEncoder().encode(value);
+    }
+
+    /**
+     * Decode the provided byte array.
+     * <p>
+     * If the input cannot be coverted, an empty array is returned
+     */
+    private static byte[] decodeOrEmptyBytes(final byte[] bytes) {
+        try {
+            return getMimeDecoder().decode(bytes);
+        } catch (final IllegalArgumentException e) {
+            return EMPTY_BYTES;
+        }
+    }
+
+    /**
+     * Provide methods to remap values from url-safe Bas64 to standard encoding. The commons-codec
+     * implementation accepts both variants in the same input. The jdk implementaion does not, so
+     * inputs need to be
+     */
+    static class UrlByteRemapper {
```

Comment:
@ok2c Do you we really need this URL safety?

Reply:
@j3graham @michael-o I think we can drop it and re-introduce it only when needed.

Reply:
it's dropped